### ISSUE TITLE
Navigation keys "bounce" after running external script

### ISSUE THREAD

**Issue by nickwynja** [kdheepak/taskwarrior-tui#189]
**Describe the bug**

After I run an external script using the custom keyboard shortcuts, the first navigation keypress is "undone" and jumps back to where I was.

In this gif, you can see I run a shortcut by pressing `9`. The script completes, then I hit `j`, and the tui moves to the next item momentarily before "bouncing" back to the last item.

![Kapture 2021-04-26 at 15 50 37](https://user-images.githubusercontent.com/1012017/116141870-3f7f7900-a6a7-11eb-99cb-780823932257.gif)

If I hold down the `j` key, it moves several items and then jumps back all the way.

![Kapture 2021-04-26 at 15 56 54](https://user-images.githubusercontent.com/1012017/116142585-11e6ff80-a6a8-11eb-9ae9-97c62f42be1b.gif)

**To Reproduce**

- [x] Reproducible using the test data located here: <https://github.com/kdheepak/taskwarrior-testdata/>

Add the following to `taskwarrior-testdata/demo.taskrc`:

```
uda.taskwarrior-tui.shortcuts.9=sleep 5
```

Open tui and press `9` to run the script. The script executes and then hit `j` to move down. After a bit you jump back to where you were:

![Kapture 2021-04-26 at 16 04 42](https://user-images.githubusercontent.com/1012017/116143487-34c5e380-a6a9-11eb-9119-8ca3ec19da53.gif)

**Environment (please complete the following information):**

- Operating System: Mac
- Installation: local build
- taskwarrior-tui version: 0.13.12

```bash
taskwarrior-tui --version
```
- task version: 2.5.1

```bash
task --version
```

**Additional context and information**

**Comment by kdheepak** [comment]
Thanks for such a detailed issue report! It's easy to see the error in my code when you write such comprehensive bug reports. 

Also, I'm amazed you find all this haha. Reproducing this one must have been annoying to figure out! Thank you again, I appreciate it. This is fixed in the referenced PR. I'll make a new release shortly.

**Comment by nickwynja** [comment]
Glad you were able to track this one down. It was definitely a strange one!

**Comment by kdheepak** [comment]
I added in a feature where if you changed the priority and if the task list was shuffled, then `taskwarrior-tui` would try to move the selection to the task that you were previously at. However, when running shortcuts, because it was taking a few seconds, any key presses you made before that were "in the queue" to be handled while the UI blocked. After the key presses were handled, during the next "tick" `taskwarrior-tui` jumped the selection back to the previous selection. 

In the fix, I changed it such that if a key is pressed that changes the selection, discard / ignore the previous selection. This means it may be behave incorrectly in the case where if you changed priority in a task shortcut that took a long time to run and you hit `j` multiple times while the UI was blocked. For now, I'm ignoring the issues that might arise in that corner case.

Ideally, the UI would never block :) But that will require a major rewrite because of the way everything is implemented at the moment.
